csrc/device_lower/validation.cpp

```diff
@@ -579,9 +579,10 @@ void validateAndCollectVectorizeInfo(Fusion* fusion) {
       }
     }
     if (has_vectorize_dim) {
+      Expr* def = tv->definition();
       NVF_ERROR(
-          tv->definition() == nullptr || tv->definition()->isA<LoadStoreOp>() ||
-              tv->definition()->isA<SliceOp>(),
+          def == nullptr || def->isA<LoadStoreOp>() ||
+              def->isA<SliceOp>() || (def->isA<ReductionOp>()), // && def->as<ReductionOp>()->SerialGridReductionRequested(),
```

Comment:
Is the `&&` part supposed to be commented out?

Reply:
That needs to be uncommented once #1456 is merged, since that introduces the method `SerialGridReductionRequested()`. For now it is just allowing reductions.

I am starting to doubt using vectorize for this though. It would be cleaner to use `Group` for the vectorized reduction output. Then as you say, if I lower to kernel IR instead of using a helper function, we could convert that to vectorized LoadStoreOps with `CacheOp::Global` then the default validation checks would work out I think. I will mark this draft while I give that a shot.

Reply:
> lower to kernel IR instead of using a helper function

Just want to make sure that may be desirable but not necessary. Prototyping is typically easier with helper runtime functions than generating in KIR>

Reply:
I think it is desirable, since we'd get a lot of stuff for free like vectorization, inserting allocations, and hoisting predicates. I've been trying a bit to get it working as a single lowering pass but generating IR for the work buffer is challenging because it appears we need to introduce a cycle into the graph; we load from the work buffer then write back to it. We could try making the input and output global work buffers into separate `TensorView`s but we'd then need to figure out how to force those to be aliased.